internal/store/store.go

```diff
@@ -219,11 +219,14 @@ func WithAuthentication(c CredentialsGetter) Option {
 	return func(s *Store) error {
 		s.username = c.PublicAPIKey()
 		s.password = c.PrivateAPIKey()
-		t, err := c.Token()
-		if err != nil {
-			return err
+
+		if s.username == "" && s.password == "" {
+			t, err := c.Token()
+			if err != nil {
+				return err
+			}
+			s.accessToken = t
```

Comment:
we use the access token only when programmatic keys are not set

Reply:
yes, correct. I am just waiting to see if I should remove the logic that remove the auth tokens in the config commands

Reply:
This is not the behavior that I see though. If I have an invalid access token and the programmatic key I get this error without this change. I believe `t, err := c.Token()` may the root cause here

```
./bin/atlas project ls -P default-access-token
Error: token contains an invalid number of segments
```

Reply:
well that may be your own fault, `invalid number of segments` means you manually modified that token and the only case I would expect `c.Token()` to error, if you give it a random string and not a valid JWT, and by valid in the generic is a JWT token kind of way, you can verify your string in this page https://jwt.io/, if that page can parse then this function can as well

Reply:
I see. I don't think we should parse the access token if we are not using it though

Reply:
and I don't think users should be manually editing their tokens, in general my comment in this particular change is not blocking, just saying there's not much gain but it doesn't hurt, deleting things without previous warning does hurt and a lot